### PR TITLE
Warn user if they try to apply on an object without the annotation

### LINKED ISSUE #36620: Issue a warning or error when using apply on a resource lacking the LastAppliedConfig annotation
We should add `LastAppliedConfig` annotation when using `kubectl create`.
Users are keep complaining about why `kubectl apply` don't delete what they expected to be deleted.
The users think this is a bug.
See #35134 and #36509.

cc: @bgrant0607 @pwittrock

### PR DESCRIPTION
**Pull request by mengqiy**
Give user a warning when using `kubectl apply` on a resource lacking the `LastAppliedConfig` annotation

fixes #36620

Ref: #35134 and #36509.

**Release note**:
```release-note
Issue a warning when using `kubectl apply` on a resource lacking the `LastAppliedConfig` annotation
```

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36672)

### CODE REVIEW

File: pkg/kubectl/cmd/apply.go
Comment (janetkuo) [inline pkg/kubectl/cmd/apply.go]: The moment the user see this warning, it's already too late. Therefore, we should tell them what to expect in this case, or show them how to react.  

We should probably generate an error instead, to stop the users from doing this. Is this considered a breaking change? @kubernetes/kubectl

**Review by janetkuo — CHANGES_REQUESTED** [review]

File: pkg/kubectl/cmd/apply.go
Comment (ghodss) [inline pkg/kubectl/cmd/apply.go]: Yes, it would be breaking. And if this were implemented as an error, there should be a dedicated `kubectl apply`-specific command-line flag which allows you to always continue anyways and just generate a new annotation in the event that there isn't one already present.

**Review by janetkuo — APPROVED** [review]

**Comment by janetkuo** [comment]
This is better than no warnings at all. We can refine it later. /lgtm

**Comment by ymqytw** [comment]
> *either

Fixed the typo. May need re-apply the `lgtm` label if the bot removes it.

**Comment by ymqytw** [comment]
@janetkuo Rebased on HEAD and fixed the unit test.

**Comment by janetkuo** [comment]
@k8s-bot cri e2e test this

**Pull request merged**
